cartograph/server/NewMapService.py

```diff
@@ -51,6 +51,69 @@ def filter_tsv(source_dir, target_dir, ids, filename):
                 popularities_writer.writerow(row)
 
 
+def gen_config(map_name):
+    """Generate the config file for a user-generated map named <map_name> and return a path to it
+
+    :param map_name: name of new map
+    :return: path to the newly-generated config file
+    """
+    # Prevent map names with special characters (for security/prevents shell injection)
+    for c in map_name:
+        assert c in ACCEPTABLE_MAP_NAME_CHARS
+
+    if not os.path.exists(USER_CONF_DIR):
+        os.makedirs(USER_CONF_DIR)
+
+    # Generate a new conf file
+    with open('./data/conf_template.txt', 'r') as conf_template_file:
+        conf_template = string.Template(conf_template_file.read())
+    config_filename = '%s.txt' % pipes.quote(map_name)
+    config_path = os.path.join(USER_CONF_DIR, config_filename)
+    assert not os.path.exists(config_path)  # Make sure no map config with this name exists
+    with open(config_path, 'w') as config_file:
+        config_file.write(conf_template.substitute(name=map_name))
+
+    return config_path
+
+
+def gen_data(map_name, articles):
+    """Generate the data files (i.e. "TSV" files) for a map named <map_name> and a string of articles <articles>.
+
+    :param map_name: name of new map
+    :param articles: list of exact titles of articles for new map
+    :return: list of article titles for which there was no exact match in the existing dataset
+    """
+    # Generate dictionary of article names to IDs
+    # TODO: is there a way to do this once (instead of once per POST)?
+    names_path = os.path.join(SOURCE_DIR, 'names.tsv')
+    name_dict = {}
+    with codecs.open(names_path, 'r') as names:
+        names_reader = csv.reader(names, delimiter='\t')
+        for row in names_reader:
+            name = unicode(row[1], encoding='utf-8')
+            name_dict[name] = row[0]
+
+    # Generate list of IDs for article names in user request
+    ids = []
+    bad_articles = []
+    for term in articles:
+        try:
```

Comment:
In the other branch (api-metrics) ids is a set(), so let's wait until the next branch gets added in